src/engine/database/mod.rs

```diff
@@ -217,18 +231,18 @@ impl TryFrom<Value> for Comparison {
 }
 
 #[async_trait]
-pub(crate) trait Transaction: Send {
+pub trait Transaction<RequestCtx: RequestContext>: Send {
```

Comment:
I think that the abstractions that we do/will provide on the facade will be sufficient that users do not need to use a Transaction directly. The reason it was made public is because I can't seem to get around a `can't leak crate-visible trait` compiler error. Open to ways around this so we can keep the trait non-public.

Reply:
This was an oversight from the fact that I exposed transaction on the facade. Should be fixed now.